packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -532,6 +566,22 @@ export class Project extends ProjectRef {
     this.addToRolePolicy(this.createLoggingPermission());
   }
 
+  /**
+   * @override
+   */
+  public validate(): string[] {
+    const ret = new Array<string>();
+    if (this.source.type === SourceType.CodePipeline) {
+      if (this._secondarySources.length > 0) {
+        ret.push('A Project with a CodePipeline Source cannot have secondary sources');
```

Comment:
Nice check!

Can we give the user something actionable to do here to correct their (most likely) problem?

Reply:
Sure - probably mentioning the `additionalInputArtifacts` / `additionalOutputArtifacts` properties of the CodePipeline Actions would be good enough here?